extensions/http-curl/client/HTTPStream.cpp

```diff
@@ -36,7 +37,7 @@ namespace minifi {
 namespace io {
 
 HttpStream::HttpStream(std::shared_ptr<utils::HTTPClient> client)
-    : http_client_(client),
+    : http_client_(std::move(client)),
```

Comment:
now we are copying the client shared_ptr when this ctor is called and then move the copy into the client_ member
wouldnt it be more readable to just use const & and then copy that into the client_ member?

same with SensorBase::WriteCallback